presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -175,9 +175,11 @@ private static TypeSignature parseRowTypeSignature(String signature, Set<String>
             else if (c == ' ') {
                 if (bracketCount == 1 && inFieldName) {
                     checkArgument(parameterStart >= 0, "Bad type signature: '%s'", signature);
-                    fieldNames.add(signature.substring(parameterStart, i));
+                    if (parameterStart < i) {
```

Comment:
I did this small modification to enable writing the types like `row(T1, T2)` (with space), otherwise it fails (it thinks that the second filed has an the name `""`.
I can revert it and use `row(T1,T2)` if you want.